### PR TITLE
Add "precompute" and "remove-unused-brs" as a cleanup in J2clOpts.

### PR DESCRIPTION
**Pull request by rluble**
This makes the cleanup the cleanup of the bodies for functions that have had constants hoisted from them more effective.

### CODE REVIEW

File: src/passes/J2CLOpts.cpp [hunk 1]
```diff
@@ -105,7 +105,9 @@ class ConstantHoister : public WalkerPass<PostWalker<ConstantHoister>> {
       // TODO: maybe we should not introduce "nop" in the first place and try
       // removing instead.
       PassRunner runner(getModule());
+      runner.add("precompute");
       runner.add("vacuum");
+      runner.add("remove-unused-brs");
```
Comment (kripken) [inline src/passes/J2CLOpts.cpp:110]: Is there a benefit to putting `remove-unused-brs` last? In general `vacuum` is used that way (but it is not a strict rule).

File: test/lit/passes/j2cl.wast [hunk 2]
```diff
@@ -151,3 +151,47 @@
...
+  ;; CHECK-NEXT:   (call $getVar1_<once>_@Zoo)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (return
+  ;; CHECK-NEXT:   (global.get $$var2@Zoo)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $getVar2_<once>_@Zoo (result i32)
+    (if (global.get $$var2@Zoo) (then (return (global.get $$var2@Zoo))))
+    (global.set $$var2@Zoo (call $getVar1_<once>_@Zoo))
+    (return (global.get $$var2@Zoo))
```
Comment (kripken) [inline test/lit/passes/j2cl.wast:195]: Please indent these more like the outputs are, that makes it easier to visually compare them. 2-space indentation is fine though.

File: src/passes/J2CLOpts.cpp (hunk 1 above)
Comment (rluble) [reply src/passes/J2CLOpts.cpp:110]: hmm.  It definitely has an effect. I extracted this out of a bigger change and what I noticed is in the bigger change I see 

```
(func $getXXX (...) 
  (return (global.get $global))
)
```

without "remove-unused-br" and

```
(func $getXXX (...) 
  (global.get $global)
)
```
if it is in.

Not really necessary though. In my bigger change I was collecting methods that had only `(global.get ...)`  but I can collect `(return (global.get ...))`  and achieve the same effect.

Comment (kripken) [reply src/passes/J2CLOpts.cpp:110]: Oh, sorry, I do agree `--remove-unused-brs` is useful. It removes `returns` like that. But can it run before the vacuum?

Comment (rluble) [reply src/passes/J2CLOpts.cpp:110]: I misunderstood. Yes it can run before. I'll update the pr.

File: test/lit/passes/j2cl.wast (hunk 2 above)
Comment (rluble) [reply test/lit/passes/j2cl.wast:203]: Done

**Review by kripken — APPROVED** [review]

**Pull request merged**